src/jwplayer/utils/scriptLoader.ts

```diff
@@ -0,0 +1,54 @@
+class ScriptLoader {
+	createScript(
+		src: string,
+		type = 'text/javascript',
+		isAsync = true,
+		node: HTMLElement | string = null,
+		parameters: Record<string, string> = {},
+		datasets: Partial<DOMStringMap> = {},
+	): HTMLScriptElement {
+		const script: HTMLScriptElement = document.createElement('script');
+
+		script.async = isAsync;
+		script.type = type;
+		script.src = src;
+
+		Object.keys(parameters).forEach((parameter) => {
+			script.setAttribute(parameter, parameters[parameter]);
+		});
+
+		Object.keys(datasets).forEach((dataset) => {
+			script.dataset[dataset] = datasets[dataset];
+		});
+
+		if (typeof node === 'string') {
+			const temp: ChildNode = document.getElementsByTagName('script')[0];
+
+			temp.parentNode.insertBefore(script, temp);
```

Comment:
Let's keep in mind that `temp.parentNode` might be `null` 😉

Reply:
yea, will fix it. good point